modules/bpi/bpi.module

```diff
@@ -148,46 +149,90 @@ function bpi_menu() {
     'file' => 'bpi.admin.inc',
   );
 
+  $menu['bpi/edit/%'] = array(
+    'title' => 'BPI content edit',
+    'access callback' => 'bpi_access',
+    'access arguments' => ['bpi content edit', 2],
+    'page callback' => '_bpi_edit_callback',
+    'page arguments' => array(2),
+    'type' => MENU_CALLBACK,
+  );
+
   return $menu;
 }
 
 /**
- * Check if current user has access to the BPI Syndication menu.
+ * Redirects the user to BPI node edit form.
+ *
+ * @param string $bid
+ *   BPI id.
+ */
+function _bpi_edit_callback($bid) {
+  if (!$node = bpi_load_node_by_bid($bid)) {
+    drupal_not_found();
+  }
+
+  $path = 'node/' . $node->nid . '/edit';
+  drupal_goto($path);
+}
+
+/**
+ * Custom access checker for BPI actions.
+ *
+ * BPI overview:
+ * The logged in user must both have the
+ * "bpi syndicate content" permission and access to create content from BPI.
+ * BPI edit link:
+ * The logged in user must have access to the target node. This is the place
+ * where BPI id parameter is used.
```

Comment:
This function `bpi_access()` is basicly one big switch statement with no shared logic and no shared arguments. If new permissions are added the function will only grow larger.

Please split it up into multiple smaller function e.g. `bpi_access_content_syndicate()` and `bpi_access_content_edit()`.